coding-guidelines.md

```diff
@@ -0,0 +1,25 @@
+# GraphQL Java Coding guidelines used in GraphQL Java
+
+
+## General principles
+
+- We prefer closer to zero dependencies. Dont bring in guava, apache-commons, spring-xxx, y or z however much some StringUtils method might be useful. Less dependencies makes graphql-java more applicable to everyone
+
+- We prefer staying out of the HTTP stack. We are the low level engine of running graphql queries. Other concerns such as JSON and HTTP are handled better by other layers
+
+- We prefer simple code to clever code. It should be readable with well named methods. Clever nested streams and lambdas are not our thing.
+
+- We prefer general to specific. So the code should be generally applicable to use cases, not highly specific to just some use cases, even if it takes more setup.
+
+
+## more specific topics
+
+- all tests are written in spock
+
+- Optional vs Null: TBD how to move forward
```

Comment:
I think we retain null-ness (for consistency) OR we start to make the move progressively

Because there are MANY MANY places that return null in the code base

Reply:
I agree with keeping the old ones, the question is more: what is the pattern for new code?